src/main.rs

```diff
@@ -70,36 +49,42 @@ fn main() {
 }
 
 fn run() -> Result<(), MainError> {
-    let args: Vec<String> = std::env::args().collect();
-    match &*args[1] {
+    let cli = Cli::parse();
+    match &*cli.name[0] {
         "run" | "run-script" => {
-            let manifest = "package.json"
+            let mut cwd = env::current_dir().expect("Couldn't find the current working directory");
+            if cli.workspace_root {
+                cwd = workspace::find_workspace_root(&cwd)?;
+            }
+            let manifest_path = cwd.join("package.json");
+            let manifest = manifest_path
                 .pipe(File::open)
                 .map_err(MainError::from_dyn)?
                 .pipe(serde_json::de::from_reader::<_, NodeManifest>)
                 .map_err(MainError::from_dyn)?;
-            let name = args[2].clone();
+            let name = cli.name[1].clone();
             if let Some(command) = manifest.scripts.get(&name) {
                 eprintln!("> {:?}", command);
-                run_script(name, command.clone())
+                run_script(name, command.clone(), &cwd)
             } else {
                 PnError::MissingScript { name }
                     .pipe(MainError::Pn)
                     .pipe(Err)
             }
         }
-        "install" | "i" | "update" | "up" => pass_to_pnpm(&args[1..]),
-        _ => pass_to_sub(args[1..].join(" ")),
+        "install" | "i" | "update" | "up" => pass_to_pnpm(&cli.name[0..]),
+        _ => pass_to_sub(cli.name[0..].join(" ")),
     }
 }
 
-fn run_script(name: String, command: String) -> Result<(), MainError> {
+fn run_script(name: String, command: String, cwd: &dyn AsRef<Path>) -> Result<(), MainError> {
     let mut path_env = OsString::from("node_modules/.bin");
     if let Some(path) = env::var_os("PATH") {
         path_env.push(":");
         path_env.push(path);
```

Comment:
Right now, we push `node_modules/.bin` to last of `PATH`, which means that npm commands cannot override system command.

If `npm` and `pnpm` don't do the same (i.e. they push `node_modules/.bin` to the start of `PATH`), we should change this to comply with the "specs".

Reply:
Issue created https://github.com/pnpm/pn/issues/17